### PR TITLE
Update hpc-slurm-legacy-sharedvpc example and references to use Slurm V6

### PR DESCRIPTION
**Pull request by harshthakkar01**
In this PR, I used 1 host project 1 service project and then set up shared VPC in host project. Also, set up firewall rule to allow access (ssh) to this shared VPC network. Below are the result on the login, controller, compute node.

* Login node https://screenshot.googleplex.com/3c8LzgwraF8soir
```
Last login: Wed Jan 24 18:58:43 2024 from 35.235.240.0
[ext_harshthakkar_google_com@hpcsmallsh-login-login-001 ~]$ df -h
Filesystem                       Size  Used Avail Use% Mounted on
devtmpfs                         7.9G     0  7.9G   0% /dev
tmpfs                            7.9G     0  7.9G   0% /dev/shm
tmpfs                            7.9G  8.6M  7.9G   1% /run
tmpfs                            7.9G     0  7.9G   0% /sys/fs/cgroup
/dev/sda2                         50G   16G   35G  32% /
/dev/sda1                        200M  5.8M  195M   3% /boot/efi
tmpfs                            1.6G     0  1.6G   0% /run/user/0
10.43.128.2:/nfsshare           1007G     0  956G   0% /home
hpcsmallsh-controller:/opt/apps   50G   16G   35G  32% /opt/apps
tmpfs                            1.6G     0  1.6G   0% /run/user/3672659445
[ext_harshthakkar_google_com@hpcsmallsh-login-login-001 ~]$ 
```

* Controller Node
```
Last login: Wed Jan 24 18:58:06 2024 from 35.235.240.3
[ext_harshthakkar_google_com@hpcsmallsh-controller ~]$ df -h
Filesystem             Size  Used Avail Use% Mounted on
devtmpfs               7.9G     0  7.9G   0% /dev
tmpfs                  7.9G     0  7.9G   0% /dev/shm
tmpfs                  7.9G  8.6M  7.9G   1% /run
tmpfs                  7.9G     0  7.9G   0% /sys/fs/cgroup
/dev/sda2               50G   16G   35G  32% /
/dev/sda1              200M  5.8M  195M   3% /boot/efi
tmpfs                  1.6G     0  1.6G   0% /run/user/0
10.43.128.2:/nfsshare 1007G     0  956G   0% /home
tmpfs                  1.6G     0  1.6G   0% /run/user/3672659445
[ext_harshthakkar_google_com@hpcsmallsh-controller ~]$
```

* Compute node after running slurm job.
```
ext_harshthakkar_google_com@hpcsmallsh-comput-0 ~]$ 
[ext_harshthakkar_google_com@hpcsmallsh-comput-0 ~]$ df -h
Filesystem                       Size  Used Avail Use% Mounted on
devtmpfs                         119G     0  119G   0% /dev
tmpfs                            119G     0  119G   0% /dev/shm
tmpfs                            119G  8.7M  119G   1% /run
tmpfs                            119G     0  119G   0% /sys/fs/cgroup
/dev/sda2                         50G   17G   34G  34% /
/dev/sda1                        200M  5.8M  195M   3% /boot/efi
tmpfs                             24G     0   24G   0% /run/user/0
tmpfs                             24G     0   24G   0% /run/user/3672659445
hpcsmallsh-controller:/opt/apps   50G   16G   35G  32% /opt/apps
10.43.128.2:/nfsshare           1007G     0  956G   0% /home
[ext_harshthakkar_google_com@hpcsmallsh-comput-0 ~]$ 
```

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: community/examples/hpc-slurm-legacy-sharedvpc.yaml [hunk 1]
```diff
@@ -55,46 +55,52 @@ deployment_groups:
...
 
-  # This debug_partition will work out of the box without requesting additional GCP quota.
+  - id: debug_nodeset
+    source: community/modules/compute/schedmd-slurm-gcp-v6-nodeset
+    use: [network1]
+    settings:
+      node_count_dynamic_max: 4
+      machine_type: n2-standard-2
+      enable_placement: false  # the default is: true
+      disable_public_ips: false
```
Comment (mr0re1) [inline community/examples/hpc-slurm-legacy-sharedvpc.yaml:65]: Why did you need to enable public ips ?

File: community/examples/hpc-slurm-legacy-sharedvpc.yaml [hunk 2]
```diff
@@ -55,46 +55,52 @@ deployment_groups:
...
 
-  # This compute_partition is far more performant than debug_partition but may require requesting GCP quotas first.
-  - id: compute_partition
-    source: community/modules/compute/SchedMD-slurm-on-gcp-partition
-    use:
-    - network1
-    - homefs
+  - id: compute_nodeset
+    source: community/modules/compute/schedmd-slurm-gcp-v6-nodeset
+    use: [network1]
```
Comment (mr0re1) [inline community/examples/hpc-slurm-legacy-sharedvpc.yaml:77]: homefs?

File: community/examples/hpc-slurm-legacy-sharedvpc.yaml (hunk 1 above)
Comment (harshthakkar01) [reply community/examples/hpc-slurm-legacy-sharedvpc.yaml]: Removed.

File: community/examples/hpc-slurm-legacy-sharedvpc.yaml
Comment (nick-stroud) [inline community/examples/hpc-slurm-legacy-sharedvpc.yaml:1]: Have you tested controller, login, and compute with shared vpc. I see there is a setting of `shared_vpc_host_project` that is no longer used. Is something similar needed. Please document these things in the pr description.

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: community/examples/hpc-slurm-legacy-sharedvpc.yaml (hunk 1 above)
Comment (harshthakkar01) [reply community/examples/hpc-slurm-legacy-sharedvpc.yaml:64]: During manual testing I realized it seems it cannot ping DNS server if public ip is disabled.

File: community/examples/hpc-slurm-legacy-sharedvpc.yaml
Comment (harshthakkar01) [reply community/examples/hpc-slurm-legacy-sharedvpc.yaml:1]: Added the testing details in the PR description.

File: examples/README.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ md_toc github examples/README.md | sed -e "s/\s-\s/ * /"
   * [hpc-slurm-chromedesktop.yaml](#hpc-slurm-chromedesktopyaml--) ![community-badge] ![experimental-badge]
   * [flux-cluster](#flux-clusteryaml--) ![community-badge] ![experimental-badge]
   * [tutorial-fluent.yaml](#tutorial-fluentyaml--) ![community-badge] ![experimental-badge]
-  * [hpc-slurm-legacy-sharedvpc.yaml](#hpc-slurm-legacy-sharedvpcyaml--) ![community-badge] ![deprecated-badge]
+  * [hpc-slurm-legacy-sharedvpc.yaml](#hpc-slurm-legacy-sharedvpcyaml--) ![community-badge]
```
Comment (nick-stroud) [inline examples/README.md:44]: 1) I bet the link is now broken. s/#hpc-slurm-legacy-sharedvpcyaml--/#hpc-slurm-legacy-sharedvpcyaml-/
2) Move up in the list above all other experimental examples

File: community/examples/hpc-slurm-legacy-sharedvpc.yaml (hunk 1 above)
Comment (nick-stroud) [reply community/examples/hpc-slurm-legacy-sharedvpc.yaml:64]: @harshthakkar01, is this new to V6 or did V5 also have public IPs and the default just changed?

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: community/examples/hpc-slurm-legacy-sharedvpc.yaml (hunk 1 above)
Comment (harshthakkar01) [reply community/examples/hpc-slurm-legacy-sharedvpc.yaml:64]: Previously, it was using V4 and all I can see is the default is to enable public ip for compute nodes.
https://github.com/SchedMD/slurm-gcp/blob/v4/tf/modules/compute/io.tf#L35

File: examples/README.md (hunk 3 above)
Comment (harshthakkar01) [reply examples/README.md]: 1) I am not sure what you mean here. It should point to https://github.com/GoogleCloudPlatform/hpc-toolkit/tree/main/examples#hpc-slurm-legacy-sharedvpcyaml-- one right ? I just removed the deprecated-badge. Not sure if this broke the link.

2) Done. Thanks for catching this.

File: community/examples/hpc-slurm-legacy-sharedvpc.yaml (hunk 1 above)
Comment (nick-stroud) [reply community/examples/hpc-slurm-legacy-sharedvpc.yaml:64]: While that was the default in the slurm repo, our wrapper would overwrite that value to disable public ip's by default: https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/fcdc5e5e484f2a2cea86a7ff9f13a9bdea969ee5/community/modules/scheduler/SchedMD-slurm-on-gcp-controller/variables.tf#L102-L105  

It seems to me like this is some sort of regression.

File: examples/README.md (hunk 3 above)
Comment (nick-stroud) [reply examples/README.md]: The anchor (`#hpc-slurm-legacy-sharedvpcyaml--`) is generated from the contents of L983. You have changed the contents of that line so the anchor will change.

The `-` are stand in for the badges, there used to be 2 now there is just 1, you still have 2x `-` in the anchor.

File: community/examples/hpc-slurm-legacy-sharedvpc.yaml (hunk 1 above)
Comment (harshthakkar01) [reply community/examples/hpc-slurm-legacy-sharedvpc.yaml]: Followed up on this one. We need to set up NAT and IAP first if external IP is disabled. Also added these instruction in documentation.

File: examples/README.md (hunk 3 above)
Comment (harshthakkar01) [reply examples/README.md]: Updated this.

File: examples/README.md [hunk 4]
```diff
@@ -999,12 +999,14 @@ See [README](../community/examples/flux-framework/README.md)
...
 [flux-cluster.yaml]: ../community/examples/flux-framework/flux-cluster.yaml
 
-### [hpc-slurm-legacy-sharedvpc.yaml] ![community-badge] ![deprecated-badge]
+### [hpc-slurm-legacy-sharedvpc.yaml] ![community-badge]
 
 This blueprint demonstrates the use of the Slurm and Filestore modules in
 the service project of an existing Shared VPC.  Before attempting to deploy the
 blueprint, one must first complete [initial setup for provisioning Filestore in
-a Shared VPC service project][fs-shared-vpc].
+a Shared VPC service project][fs-shared-vpc]. Also since this blueprint doesn't
```
Comment (nick-stroud) [inline examples/README.md:1007]: I would also like to mention the need for firewall rules. Maybe something like:

"Depending on how the shared VPC was created one may have to perform a few additional manual steps to configure the VPC. One may need to create firewall rules allowing SSH to be able to access the controller and login nodes. Also since this blueprint doesn't use external ips..."

File: examples/README.md [hunk 5]
```diff
@@ -999,12 +999,14 @@ See [README](../community/examples/flux-framework/README.md)
...
+use external ip for compute nodes, one must need to [set up cloud nat][cloudnat]
```
Comment (nick-stroud) [inline examples/README.md:1008]: Grammer:
s/use external ip for compute nodes/use external IPs for the compute nodes/

**Review by nick-stroud — APPROVED** [review]

File: examples/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (harshthakkar01) [reply examples/README.md]: Done.

Comment (harshthakkar01) [reply examples/README.md]: Done.

File: examples/README.md (hunk 4 above)
Comment (harshthakkar01) [reply examples/README.md]: Done
